KOGITO-5493 process definition table & process form

AjayJagan opened the pull request:
- Add process definition table - https://issues.redhat.com/browse/KOGITO-5868
 - Add process form page - https://issues.redhat.com/browse/KOGITO-5870
 - Add process form component(uniforms) - https://issues.redhat.com/browse/KOGITO-5869
 - get process definition using OpenAPI - https://issues.redhat.com/browse/KOGITO-6275

Sara4994 commented on ui-packages/packages/components-common/package.json:
```diff
@@ -30,7 +30,12 @@
     "babel-loader": "8.1.0",
-    "react-docgen-typescript-loader": "3.7.2"
+    "react-docgen-typescript-loader": "3.7.2",
+    "@types/json-schema": "^7.0.7",
```
remove ^

Sara4994 commented on ui-packages/packages/components-common/src/components/FormFooter/FormFooter.tsx:
```diff
@@ -23,19 +23,21 @@ import { componentOuiaProps, OUIAProps } from '@kogito-apps/ouia-tools';
 interface IOwnProps {
   actions?: FormAction[];
   enabled?: boolean;
+  formApiRef?: any;
```
add  type in place of any

Sara4994 commented on ui-packages/packages/components-common/src/components/FormFooter/FormFooter.tsx:
```diff
@@ -44,7 +46,12 @@ const FormFooter: React.FC<IOwnProps & OUIAProps> = ({
-    <ActionList {...componentOuiaProps(ouiaId, 'form-footer', ouiaSafe)}>
+    <ActionList
+      {...componentOuiaProps(ouiaId, 'form-footer', ouiaSafe)}
+      style={{
```
add a className instead of inline styles

Sara4994 commented on ui-packages/packages/components-common/src/components/FormRenderer/FormRenderer.tsx:
```diff
@@ -24,7 +24,7 @@ import FormFooter from '../FormFooter/FormFooter';
   formSchema: any;
   model?: any;
-  onSubmit?: (data: any) => void;
+  onSubmit?: (data: any, formApiRef: any) => void;
```
same here, please mention type

Sara4994 commented on ui-packages/packages/jobs-management/webpack.common.js:
```diff
@@ -93,7 +96,10 @@ module.exports = {
           path.resolve(
-            '../../node_modules/@kogito-apps/management-console-shared/dist/src/static'
+            '../../node_modules/@kogito-apps/management-console-shared/dist/src/static',
+            path.resolve(
```
this path.resolve looks nested, could be placed individually

Sara4994 commented on ui-packages/packages/process-definition-list/src/envelope/components/ProcessDefinitionList/ProcessDefinitionList.tsx:
```diff
@@ -0,0 +1,118 @@
+      setError(errorContent);
+    }
+  };
+  const [columns] = useState<DataTableColumn[]>([
```
This can be converted normal variable or an array, since we are not using setState method here

Sara4994 commented on ui-packages/packages/process-definition-list/src/envelope/components/ProcessDefinitionList/ProcessDefinitionList.tsx:
```diff
@@ -0,0 +1,118 @@
+  >([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [filterProcessNames, setFilterProcessNames] = useState<string[]>([]);
+  const [error, setError] = useState<any>(null);
```
string type?

Sara4994 commented on ui-packages/packages/process-definition-list/src/envelope/components/ProcessDefinitionList/ProcessDefinitionList.tsx:
```diff
@@ -0,0 +1,118 @@
+    await driver.setProcessDefinitionFilter(filterProcessNames);
+  };
+
+  const filterProcessDefinition = () => {
```
return type mising

Sara4994 commented on ui-packages/packages/process-definition-list/src/envelope/components/utils/ProcessDefinitionListUtils.tsx:
```diff
@@ -0,0 +1,31 @@
+};
+
+export const getActionColumn = (
+  startProcess: (processDefinition: any) => void
```
specific type in place of any

Sara4994 commented on ui-packages/packages/process-form/src/api/ProcessFormChannelApi.ts:
```diff
@@ -0,0 +1,24 @@
+export interface ProcessFormChannelApi {
+  processForm__getProcessFormSchema(
+    processDefinitionData: ProcessDefinition
+  ): Promise<any>;
```
Promise<void>?

AjayJagan replied:
it returns a promise of type any , if I am correct

Sara4994 commented on ui-packages/packages/process-form/src/api/ProcessFormDriver.ts:
```diff
@@ -0,0 +1,25 @@
+ * Interface that defines a Driver for ProcessForm views.
+ */
+export interface ProcessFormDriver {
+  getProcessFormSchema(processDefinitionData: ProcessDefinition): Promise<any>;
```
same here void?

AjayJagan replied:
since it returns a process schema it can return any

pefernan replied:
I'd do Record<string, any> to be consistent with task form

Sara4994 commented on ui-packages/packages/process-form/src/api/ProcessFormEnvelopeApi.ts:
```diff
@@ -0,0 +1,32 @@
+export interface ProcessFormEnvelopeApi {
+  processForm__init(
+    association: Association,
+    processDefinition: any
```
type?

Sara4994 commented on ui-packages/packages/process-form/src/embedded/EmbeddedProcessFormChannelApiImpl.ts:
```diff
@@ -0,0 +1,36 @@
+
+  processForm__getProcessFormSchema(
+    processDefinitionData: ProcessDefinition
+  ): Promise<any> {
```
void?

AjayJagan replied:
since it returns a process schema it can return any

pefernan replied:
Record<string, any>

Sara4994 commented on ui-packages/packages/process-form/src/envelope/components/ProcessForm/ProcessForm.tsx:
```diff
@@ -0,0 +1,103 @@
+}) => {
+  const [processFormSchema, setProcessFormSchema] = useState<any>({});
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState<any>(null);
```
string?

Sara4994 commented on ui-packages/packages/process-form/src/envelope/components/ProcessForm/ProcessForm.tsx:
```diff
@@ -0,0 +1,103 @@
+    init();
+  }, [isEnvelopeConnectedToChannel]);
+
+  const init = async () => {
```
"Promise <void>"

Sara4994 commented on ui-packages/packages/process-form/src/envelope/ProcessFormEnvelopeViewDriver.ts:
```diff
@@ -0,0 +1,41 @@
+    private readonly channelApi: MessageBusClientApi<ProcessFormChannelApi>
+  ) {}
+
+  getProcessFormSchema(processDefinitionData: ProcessDefinition): Promise<any> {
```
void?

AjayJagan replied:
since it returns a form schema it can be any

pefernan replied:
Record<string, any>

Sara4994 commented on ui-packages/packages/process-form/src/envelope/components/ProcessForm/ProcessForm.tsx:
```diff
@@ -0,0 +1,103 @@
+    try {
+      await driver.startProcess(value);
+      formApiRef.reset();
+    } catch (e) {
```
Catch statement is empty here

Sara4994 commented on ui-packages/packages/process-form/src/envelope/components/ProcessForm/tests/mocks/ConfirmTravelForm.ts:
```diff
@@ -0,0 +1,93 @@
+ * limitations under the License.
+ */
+
+/*
```
duplicate header

Sara4994 commented on ui-packages/packages/runtime-tools-dev-ui-webapp/package.json:
```diff
@@ -41,13 +42,17 @@
   "devDependencies": {
-    "core-js": "3.6.5"
+    "core-js": "3.6.5",
+    "swagger-ui-express": "^4.2.0"
```
please remove ^

Sara4994 commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/channel/apis/apis.tsx:
```diff
@@ -332,3 +335,74 @@ export const saveFormContent = (
+
+export const getProcessSchema = (
+  processDefinitionData: ProcessDefinition
+): Promise<any> => {
```
type?

pefernan replied:
Record<string, any>

Sara4994 commented on ui-packages/packages/components-common/src/components/FormRenderer/FormRenderer.tsx:
```diff
@@ -53,19 +54,34 @@ const FormRenderer: React.FC<IOwnProps & OUIAProps> = ({
+    <React.Fragment>
+      {' '}
+      <div
+        style={{
```
please avoid inline styles.

Sara4994 commented:
@AjayJagan Good work :clap: PR looks great and huge... I have added a few comments related to types, styles etc. It will be good if we add types wherever it is missed(includes variables, function return types, useState etc). I will test the functionality shortly.

pefernan commented on ui-packages/packages/process-definition-list/src/envelope/components/ProcessDefinitionList/ProcessDefinitionList.tsx:
```diff
@@ -0,0 +1,118 @@
+}
+const ProcessDefinitionList: React.FC<ProcessDefinitionListProps &
+  OUIAProps> = ({ isEnvelopeConnectedToChannel, driver, ouiaId, ouiaSafe }) => {
+  const [processDefinitionList, SetProcessDefinitionList] = useState<
```
`setProcessDefinitionList`

pefernan commented on ui-packages/packages/process-definition-list/src/envelope/components/ProcessDefinitionListToolbar/ProcessDefinitionListToolbar.tsx:
```diff
@@ -0,0 +1,157 @@
+  Tooltip
+} from '@patternfly/react-core';
+import { FilterIcon, SyncIcon } from '@patternfly/react-icons';
+import _ from 'lodash';
```
Try importing only the individual methods from `lodash` instead of the full object.

pefernan commented on ui-packages/packages/process-definition-list/src/envelope/components/utils/ProcessDefinitionListUtils.tsx:
```diff
@@ -0,0 +1,31 @@
+    path: 'actions',
+    bodyCellTransformer: (value, rowData: ProcessDefinition) => (
+      <div onClick={() => startProcess(rowData)}>
+        <Tooltip content="start process">
```
Uppercase here! `Start new process`

pefernan commented on ui-packages/packages/process-form/src/embedded/EmbeddedProcessFormChannelApiImpl.ts:
```diff
@@ -0,0 +1,36 @@
+    return this.driver.getProcessFormSchema(processDefinitionData);
+  }
+
+  processForm__startProcess(formJSON: any): Promise<void> {
```
try a better variable name, `data` or something like that, `formJSON`isn't clear

pefernan commented on ui-packages/packages/runtime-tools-dev-ui-webapp/server/MockData/graphql.js:
```diff
@@ -4138,5 +4138,15 @@ UserTaskInstances: [
       'http://localhost:4000/hiring/b4096227-2c8a-463f-a7a9-776027f7123422/HRInterview/5fe852de-8d00-4197-9936-3842c648fe123422'
   }
+],
+ProcessDefinitions:[
```
maybe we don't need this here. Would it be possible to have them on the controller, so we can just have those defs too? we may have an error opening the form for any of this, so we can see the error state.

AjayJagan replied:
what I have done is .. I have used a mock swagger endpoint in mock server, so that is where I get endpoint form. So what we can do is to simulate an error , I can do something like add a condition(create a business key with error) and then return a 500 or 404 error to see the error notification .. wdyt

pefernan commented on ui-packages/packages/runtime-tools-dev-ui-webapp/server/MockData/types.js:
```diff
@@ -33,6 +33,9 @@ module.exports = typeDefs = gql`
       orderBy: JobOrderBy
       pagination: Pagination
       ): [Job]
+    ProcessDefinitions(
```
we may not need this too..

pefernan commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/channel/ProcessForm/ProcessFormGatewayApi.ts:
```diff
@@ -0,0 +1,56 @@
+  }
+
+  startProcess(
+    formJSON: any,
```
this name...

pefernan commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/channel/apis/apis.tsx:
```diff
@@ -332,3 +335,74 @@ export const saveFormContent = (
+};
+
+export const startProcessInstance = (
+  formJSON: any,
```
this name

pefernan commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/components/pages/ProcessFormPage/ProcessFormPage.tsx:
```diff
@@ -0,0 +1,132 @@
+          title={`Start ${processDefinition.processName}`}
+          extra={<InlineEdit />}
+        />
+      </PageSection>
```
I'd move the notification under the title, to keep it consistent with the task form page. There are also some problems here. There is a lot of space between the header and the form, probably that's beause of the notification section. Also there's a double vertical scroll.

![process-form](https://user-images.githubusercontent.com/2429832/145471729-9db765dc-5af2-43a2-abba-8d5fe28d935d.png)

pefernan commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/components/pages/ProcessFormPage/components/InlineEdit/InlineEdit.tsx:
```diff
@@ -0,0 +1,88 @@
+          className="pf-c-inline-edit__value"
+          id="single-editable-example-label"
+        >
+          {gatewayApi.getBusinessKey().length > 0
```
Nippiccking here... but would it be possible to show the default value (in empty state) as disabled?

pefernan reviewed (changes requested):
@AjayJagan this looks great! I added some comments inline to fix some small issues I found.

Also noticed that the task form page is looking weird after this change:
![task-form](https://user-images.githubusercontent.com/2429832/145472239-ca8a3322-afb5-413b-89c4-f2cf8e6c0ac2.png)

Please take a look there too. I think the double scroll issue is because now the form renderer has a height, I'd say that we could keep it as it was before so it takes the content height.

AjayJagan commented on ui-packages/packages/components-common/src/components/utils/FormActionsUtils.tsx:
```diff
@@ -44,7 +45,9 @@ export const convertActionsToButton = (
     // Assuming that Complete will be the default act
-    return label.toLowerCase() === 'complete';
+    return (
+      label.toLowerCase() === 'complete' || label.toLowerCase() === 'start'
```
do you think this codition is fine : 

```  const isPrimary = (label: string): boolean => {
    // Assuming that Complete will be the default act
    return (
      label.toLowerCase() === 'complete' || actions.length===1
    );
  };```

AjayJagan commented:
> @AjayJagan this looks great! I added some comments inline to fix some small issues I found. Please, rebase with master... just merged a PR and added some conflicts here :S
> 
> Also noticed that the task form page is looking weird after this change: ![task-form](https://user-images.githubusercontent.com/2429832/145472239-ca8a3322-afb5-413b-89c4-f2cf8e6c0ac2.png)
> 
> Please take a look there too. I think the double scroll issue is because now the form renderer has a height, I'd say that we could keep it as it was before so it takes the content height.

yeah you are right .. the problem is when I moved the form footer outside(autoform doesnot handle the alignment) ..so now we have to handle the alighment with a custom css

jstastny-cz commented on ui-packages/packages/process-form/src/envelope/components/ProcessForm/ProcessForm.tsx:
```diff
@@ -0,0 +1,105 @@
+  }
+
+  return (
+    <div {...componentOuiaProps(ouiaId, 'process-form-renderer', ouiaSafe)}>
```
again `!isLoading` would be a good candidate instead of ouiaSafe (or combine the two).

jstastny-cz replied:
And `process-form-renderer` -> `process-form` to reflect name of component.

jstastny-cz commented on ui-packages/packages/process-list/src/envelope/components/ProcessList/ProcessList.tsx:
```diff
@@ -286,4 +286,4 @@ const ProcessListPage: React.FC<ProcessListPageProps & OUIAProps> = ({
 };
 
-export default ProcessListPage;
+export default ProcessList;
```
Please also change to ouia-component-type for to `process-list`

jstastny-cz commented on ui-packages/packages/runtime-tools-dev-ui-webapp/resources/index.html:
```diff
@@ -40,7 +40,9 @@
     dataIndexUrl: "http://localhost:4000/graphql",
-    page: "JobsManagement"
+    page: "JobsManagement",
+    devUIUrl:"http://localhost:4000",
```
devUIUrl and openApiPath are parameterized elsewhere, is it applicable also here?

jstastny-cz commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/components/pages/ProcessFormPage/ProcessFormPage.tsx:
```diff
@@ -0,0 +1,127 @@
+    <React.Fragment>
+      <PageSection
+        variant="light"
+        {...componentOuiaProps(ouiaId, 'process-form-page-section', ouiaSafe)}
```
```suggestion
        {...componentOuiaProps('title'+ouiaId, 'process-form-page-section', ouiaSafe)}
```

jstastny-cz commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/components/pages/ProcessFormPage/ProcessFormPage.tsx:
```diff
@@ -0,0 +1,127 @@
+          </div>
+        )}
+      </PageSection>
+      <PageSection>
```
```suggestion
      <PageSection
        {...componentOuiaProps('content'+ouiaId, 'process-form-page-section', ouiaSafe)}
      >
```

jstastny-cz commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/components/pages/ProcessFormPage/components/InlineEdit/InlineEdit.tsx:
```diff
@@ -0,0 +1,97 @@
+  };
+
+  return (
+    <div
```
Since everywhere we have OUIA attributes added, could we also add here?

jstastny-cz commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/envelope/tests/RuntimeToolsDevUIEnvelopeView.test.tsx:
```diff
@@ -38,6 +38,8 @@ describe('RuntimeToolsDevUIEnvelopeView tests', () => {
         forwardRef.current.setUsers([]);
         forwardRef.current.navigateTo('test');
+        forwardRef.current.setDevUIUrl('http://localhost:8080'),
+          forwardRef.current.setOpenApiPath('/docs/openapi.json');
```
```suggestion
        forwardRef.current.setOpenApiPath('/docs/openapi.json');
```

jstastny-cz commented on ui-packages/packages/task-form/src/envelope/components/EmptyTaskForm/tests/EmptyTaskForm.test.tsx:
```diff
@@ -173,27 +173,27 @@ describe('EmptyTaskForm Test', () => {
-    releaseButton.props().onClick();
-    expect(doSubmit).toHaveBeenLastCalledWith('release');
-  });
+  // need to verify
```
please do verify (and remove or fix) before merging this :-)

jstastny-cz reviewed (commented):
Thanks, overall looks good, I have a few remarks mainly on ouia topic.

pefernan commented on runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension-deployment/src/main/resources/dev-templates/processInstances.html:
```diff
@@ -35,7 +35,9 @@
+        devUIUrl: "http://localhost:{config:property('quarkus.http.port') or '8080'}",
+        openApiPath: "{config:property('quarkus.smallrye-openapi.path') or '/q/openapi.json'}"
     })
 </script>
```
This link seems not to work fine. I think that's because you changed the `DevUIRoutes.tsx` but didn't change the page here.

AjayJagan replied:
changed names properly

pefernan commented on ui-packages/packages/process-definition-list/src/envelope/components/utils/ProcessDefinitionListUtils.tsx:
```diff
@@ -0,0 +1,31 @@
+    label: 'Actions',
+    path: 'actions',
+    bodyCellTransformer: (value, rowData: ProcessDefinition) => (
+      <div onClick={() => startProcess(rowData)}>
```
I'd make this primary and add the mouse button on hover, could it be a button instead of a icon?

pefernan reviewed (changes requested):
Just two comments, apart from that I'd say that everything else is working well! thanks!

pefernan commented on ui-packages/pom.xml:
```diff
@@ -117,7 +125,7 @@
         </configuration>
         <executions>
-          <execution>
+          <execution>form-details
```
Please fix this!

AjayJagan replied:
sorry , my bad

pefernan commented on runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension-deployment/src/main/resources/dev-templates/embedded.html:
```diff
@@ -14,9 +14,9 @@
+<a href="{urlbase}/processes" class="badge badge-light">
     <i class="fa fa-project-diagram fa-fw"></i>
-    Process Instances <span id="processInstancesCount" class="badge badge-light"/>
+    Processes <span id="processesCount" class="badge badge-light"/>
```
We need this back, here we are displaying the list of process instances in the runtime.

pefernan commented on runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension-deployment/src/main/resources/dev-templates/embedded.html:
```diff
@@ -49,7 +49,7 @@
     };
 
-    fillCount("processInstances");
+    fillCount("processes");
```
Same here, we use this string (`processes` here) to query on a specific rest endpoint on `DataIndexService` and set the value in the right badge. The endpoint is `processInstances` not `process`so the query just fails and no data is loaded into the badge.

pefernan replied:
https://github.com/kiegroup/kogito-apps/blob/main/runtime-tools-quarkus-extension-parent/runtime-tools-quarkus-extension/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/dataindex/DataIndexService.java#L67

pefernan reviewed (approved):
+1 when green! Thank you @AjayJagan great work

paulovmr commented on ui-packages/packages/management-console-webapp/webpack.common.js:
```diff
@@ -68,7 +68,10 @@ module.exports = {
           ),
           path.resolve(
-            '../../node_modules/@kogito-apps/management-console-shared/dist/static'
+            '../../node_modules/@kogito-apps/management-console-shared/dist/src/static'
```
@AjayJagan After my PR, all those paths in webpack files should be like `[...]/dist/static` and not `[...]/dist/src/static`, could you please check?

AjayJagan replied:
@paulovmr  - thanks for pointing out , I have resolved it now

jstastny-cz commented on ui-packages/packages/runtime-tools-dev-ui-webapp/src/components/pages/ProcessFormPage/tests/ProcessFormPage.test.tsx:
```diff
@@ -0,0 +1,50 @@
+      </BrowserRouter>
+    );
+
+    expect(wrapper).toMatchSnapshot();
```
Here it would be great to use `find` to cut out the BrowserRouter from the snapshot itself. `wrapper.find('ProcessFormPage')` should work.

jstastny-cz reviewed (commented):
Just two minor comments, otherwise looks good.

jstastny-cz reviewed (approved):
Looks good, thank you @AjayJagan .

Sara4994 commented:
@AjayJagan +1 from me, when build passes green.

jstastny-cz commented:
There seems to be an issue with UI packages build now, to me it's not obvious where exactly.
```
[2022-01-04T09:58:16.786Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp: PASS src/channel/ProcessDefinitionList/tests/ProcessDefinitionListGatewayApi.test.ts
[2022-01-04T09:58:17.369Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp: node:internal/process/promises:246
[2022-01-04T09:58:17.369Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp:           triggerUncaughtException(err, true /* fromPromise */);
[2022-01-04T09:58:17.369Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp:           ^
[2022-01-04T09:58:17.369Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp: [UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "TypeError: this.args.view(...).setDevUIUrl is not a function".] {
[2022-01-04T09:58:17.369Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp:   code: 'ERR_UNHANDLED_REJECTION'
[2022-01-04T09:58:17.369Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp: }
[2022-01-04T09:58:17.627Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[2022-01-04T09:58:17.627Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.@kogito-apps/runtime-tools-dev-ui-webapp: error Command failed with exit code 1.
[2022-01-04T09:58:17.627Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp: error Command failed with exit code 1.
[2022-01-04T09:58:17.627Z] [INFO] @kogito-apps/runtime-tools-dev-ui-webapp: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[2022-01-04T09:58:17.627Z] [INFO] lerna ERR! yarn run test:coverage --registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/ exited 1 in '@kogito-apps/runtime-tools-dev-ui-webapp'
[2022-01-04T09:58:17.627Z] [INFO] error Command failed with exit code 1.
[2022-01-04T09:58:17.627Z] [INFO] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

manstis commented:
Have you rebased?!?

The method does not appear to exist on `main`.

https://github.com/kiegroup/kogito-apps/blob/main/ui-packages/packages/runtime-tools-dev-ui-webapp/src/envelope/RuntimeToolsDevUIEnvelopeApiImpl.ts#L63-L65

https://github.com/kiegroup/kogito-apps/blob/main/ui-packages/packages/runtime-tools-dev-ui-webapp/src/envelope/RuntimeToolsDevUIEnvelopeViewApi.ts

@pefernan Perhaps you can advise @jstastny-cz?

AjayJagan commented:
> Have you rebased?!?
> 
> The method does not appear to exist on `main`.
> 
> https://github.com/kiegroup/kogito-apps/blob/main/ui-packages/packages/runtime-tools-dev-ui-webapp/src/envelope/RuntimeToolsDevUIEnvelopeApiImpl.ts#L63-L65
> 
> https://github.com/kiegroup/kogito-apps/blob/main/ui-packages/packages/runtime-tools-dev-ui-webapp/src/envelope/RuntimeToolsDevUIEnvelopeViewApi.ts
> 
> @pefernan Perhaps you can advise @jstastny-cz?

let me rebase then :)

pefernan commented:
El dt., 4 de gen. 2022, 8:20, kie-ci ***@***.***> va escriure:

> The (build) kogito-apps check has *failed*. Please check the logs
> <https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/413/display/redirect>
> .
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kiegroup/kogito-apps/pull/1132#issuecomment-1004578037>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AASRHCFI663H76JG5LO2CLTUUKNUZANCNFSM5IX3DE4Q>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>
